analyzers/src/SonarAnalyzer.CSharp/Rules/CommentedOutCode.cs

```diff
@@ -34,163 +34,115 @@ namespace SonarAnalyzer.Rules.CSharp
     [Rule(DiagnosticId)]
     public sealed class CommentedOutCode : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S125";
+        private const string DiagnosticId = "S125";
         private const string MessageFormat = "Remove this commented out code.";
+        private const int CommentMarkLength = 2;
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        private static readonly string[] CodeEndings = { ";", "{", ";}", "{}" };
+        private static readonly string[] CodeParts = { "++", "catch(", "switch(", "try{", "else{" };
+        private static readonly string[] CodePartsWithRelationalOperator = { "for(", "if(", "while(" };
+        private static readonly string[] RelationalOperators = { "<", ">", "<=", ">=", "==", "!=" };
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxTreeActionInNonGenerated(
-                c =>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxTreeActionInNonGenerated(c =>
                 {
                     foreach (var token in c.Tree.GetRoot().DescendantTokens())
                     {
-                        CheckTrivias(token.LeadingTrivia, c);
-                        CheckTrivias(token.TrailingTrivia, c);
+                        CheckTrivias(c, token.LeadingTrivia);
+                        CheckTrivias(c, token.TrailingTrivia);
                     }
                 });
-        }
 
-        private static void CheckTrivias(IEnumerable<SyntaxTrivia> trivias, SyntaxTreeAnalysisContext context)
+        private static void CheckTrivias(SyntaxTreeAnalysisContext context, IEnumerable<SyntaxTrivia> trivias)
         {
             var shouldReport = true;
             foreach (var trivia in trivias)
             {
                 // comment start is checked because of  https://github.com/dotnet/roslyn/issues/10003
```

Comment:
In my understanding, the issue is still relevant as we can run on old Roslyn.
The 2nd part of the `&&` is dealing with it `... StartsWith("/**",...`
Swapping this condition is a performance optimization, because `IsKind` is super cheap and `.ToFullString()` is little bit more expensive.

Reply:
You're right, either way the check is still the same. I've got confused.